wagtail/core/models.py

```diff
@@ -310,16 +321,295 @@ def get_site_root_paths():
         result = cache.get('wagtail_site_root_paths')
 
         if result is None:
-            result = [
-                (site.id, site.root_page.url_path, site.root_url)
-                for site in Site.objects.select_related('root_page').order_by(
-                    '-root_page__url_path', '-is_default_site', 'hostname')
-            ]
+            result = []
+
+            for site in Site.objects.select_related('root_page', 'root_page__locale').order_by('-root_page__url_path', '-is_default_site', 'hostname'):
+                if getattr(settings, 'WAGTAIL_I18N_ENABLED', False):
+                    result.extend([
+                        (site.id, root_page.locale.language_code, root_page.url_path, site.root_url)
```

Comment:
Taking another look at what urlparse does - it's more of a tuple with some extra helper properties for interpreting the information that's already available in the tuple items (e.g. extracting hostname and port from `netloc`). On further consideration, stashing away the language code in a hidden property (just so that existing code that unpacks it as a 3-tuple can continue to do so) would be "too much magic", and breaking backwards compatibility would be the lesser evil. So, happy to go with the current code, with an upgrade consideration note (which I'll write tomorrow when I merge).